src/greenpak/i2c.py

```diff
@@ -48,11 +45,91 @@ def read(
         """
         assert False, f"Class {self.__class__} does not implement read()"
 
+    def write_read(self, addr: int, wreq: bytearray, rbytes: int,
```

Comment:
I don't need the write_read() in the base interface. Can you delete or move to the SMB implementation?

Reply:
As we elaborated above: the smbus adapter cannot or rather should not split write, read, that was pretty much the Whole reason for this PR - to allow the implementation that works well with a transaction rather single or split reads, writes.

All you do is just to move gp_write, gp_read, lower to one level, just the steps.  The advantage for lower layer - like smbus adapter - is again, to repeat, Direct mapping to single api call to perform read operation

Thus why not allow this in the interface?

Reply:
My goal is to have a clean interface and then drive from it different implementation. The write_read transaction is not supported by the I2cAdapter nor is it required, so no need to have it in the base interface.  Having base ainterface with sg_read/write/scan provides a clean interface and at the same time allows you to implement for the SMB and for me for the I2CAdapter.